recipes/msix/all/conandata.yml

```diff
@@ -0,0 +1,10 @@
+sources:
+  "1.7":
+    url: "https://github.com/microsoft/msix-packaging/archive/refs/tags/v1.7.zip"
+    sha256: "23179d577d0b9d34e93374ae38284ab9f1247ff6cc9175851983fb114ab00087"
+patches:
+  "1.7":
```

Comment:
Those are rather huge patches. Could you please consider upstreaming them?

Reply:
I don't think the upstream is interested in our modifications because here there are mostly removals of unnecessary for CCI build stuff like calculating the version or the code related to `nuget` packaging. Is it so critical to deal with the size of our patches?

Reply:
Yes, because this is really a nightmare to maintain

Reply:
All the fixes provided by our patches fall into the following categories:
- removal of version calculation logic, including the invocations of `git describe` and parsing the obtained description string;
- removal of NuGet-related logic, including the calls of `configure_file` for creating `nuspec` and `targets` files;
- removal of hardcoded configuration logic, for example, manual configuring of the `/MD` and `/MT` flags or `set(CMAKE_POSITION_INDEPENDENT_CODE ON);`
- removal of CMake logic for the vendored libraries (openssl, xerces-c, zlib);
- addition of the `conan_basic_setup` call;

It seems quite difficult to upstream these changes, especially the changes related to the packaging and release flow (git versioning, NuGet, etc). It's unlikely that the MSIX team is going to change their release processes just because of the Conan recipe we're working on.

Anyway, if we try to rework some patches, what is the criteria to stick to? When it becomes "easy enough" to maintain?

Reply:
It should be possible to hide the NuGet/Version calculation and similar stuff behind a CMake option. The default value should then be that nothing changes for upstream

> addition of the conan_basic_setup call;

That is why we most of the time use a "CMake wrapper", which calls `conan_basic_setup` and add the upstream source directory with `add_subdirectory()`

Reply:
[Done](https://github.com/conan-io/conan-center-index/pull/6475/commits/a3144aa600399cfd7cd535b10d847ed80b658bed)